### PR TITLE
Add `std` feature and Error impl for `TinyStrError`

### PR DESCRIPTION
**Pull request by bheylin**
Fixes: #3007

### CODE REVIEW

File: utils/tinystr/src/error.rs [hunk 1]
```diff
@@ -4,6 +4,9 @@
 
 use displaydoc::Display;
 
+#[cfg(feature = "std")]
```
Comment (sffc) [inline utils/tinystr/src/error.rs:7]: Issue: it needs to be added to tinystr's Cargo.toml, too

**Review by sffc — CHANGES_REQUESTED** [review]

File: utils/tinystr/src/error.rs (hunk 1 above)
Comment (bheylin) [reply utils/tinystr/src/error.rs:7]: Ah forgot to check that in. Late night changes :roll_eyes:

**Review by sffc — APPROVED** [review]

File: utils/tinystr/src/error.rs (hunk 1 above)
Comment (bheylin) [reply utils/tinystr/src/error.rs:7]: Missing some feature propagation it seems...

**Comment by sffc** [comment]
You probably need to change the lib.rs file since you added the std feature; see here:

https://github.com/unicode-org/icu4x/blob/main/docs/process/boilerplate.md

You should be able to reproduce this locally.

**Comment by bheylin** [comment]
> You should be able to reproduce this locally.

Resolved in https://github.com/unicode-org/icu4x/pull/3009/commits/b7b156cccde971cc8b0227546d9c2c6f220ada04

I also forwarded the `std` feature through to `tinystr` from other private packages that have it as a dep.

**Comment by Manishearth** [comment]
> I also forwarded the `std` feature through to `tinystr` from other private packages that have it as a dep.

I don't think we should do this, if people want the impl they can enable it themselves. There's an argument to be made for crates that return TinyStrErrors, but aiui only icu_timezone does that and I'm ambivalent on whether it should forward the feature.

Features should try to be minimal since they're additive.

**Comment by bheylin** [comment]
> > I also forwarded the `std` feature through to `tinystr` from other private packages that have it as a dep.
> 
> I don't think we should do this, if people want the impl they can enable it themselves. There's an argument to be made for crates that return TinyStrErrors, but aiui only icu_timezone does that and I'm ambivalent on whether it should forward the feature.
> 
> Features should try to be minimal since they're additive.

Fair point, forwarding the `std` feature is also outside the scope of resolving #3007.
I'll removed the forwarding and rebase the PR.

**Review by Manishearth — APPROVED** [review]

**Pull request merged**